lib/internal/freelist.js

```diff
@@ -25,6 +27,11 @@ class FreeList {
     }
     return false;
   }
+
+  updateMax(max) {
+    validateNumber(max, 'max');
```

Comment:
Since the `validateNumber()` validator also allows floating-point numbers, I'm unsure if those should be allowed here; shouldn't this only allow integers?

Reply:
Fixed, thanks.